src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -55,6 +55,20 @@ public static bool IsSystemVoid([NotNullWhen(returnValue: true)] this ITypeSymbo
         public static bool IsNullable([NotNullWhen(returnValue: true)] this ITypeSymbol? symbol)
             => symbol?.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T;
 
+        public static bool IsNullable(
+            [NotNullWhen(returnValue: true)] this ITypeSymbol? symbol,
+            [NotNullWhen(returnValue: true)] out ITypeSymbol? underlyingType)
+        {
+            if (symbol is INamedTypeSymbol { OriginalDefinition: { SpecialType: SpecialType.System_Nullable_T } } namedType)
+            {
+                underlyingType = namedType.TypeArguments.Single();
```

Comment:
```suggestion
                underlyingType = namedType.TypeArguments[0];
```

See `GetNullableUnderlyingType` for the compiler implementation of this method, which also does direct indexing.

Reply:
WFM.